index.js

```diff
@@ -1,33 +1,47 @@
-const { extname } = require('path')
 const fs = require('fs').promises
 const { compileTemplate } = require('@vue/compiler-sfc')
 const { optimize: optimizeSvg } = require('svgo')
 
 module.exports = function svgLoader (options = {}) {
   const { svgoConfig, svgo } = options
 
+  const svgRegex = /\.svg(\?(raw|url|component))?$/
   return {
     name: 'svg-loader',
     enforce: 'pre',
-
+    resolveid (id) {
+      return id.match(svgRegex) ? id : null
+    },
     async load (id) {
-      const [path, parameter] = id.split('?')
-
-      if (!extname(path).startsWith('.svg') || parameter === 'url') {
-        return null
+      if (!id.match(svgRegex)) {
+        return
       }
-
-      const svg = await fs.readFile(path, 'utf-8')
-
-      const optimizedSvg = svgo === false ? svg : optimizeSvg(svg, svgoConfig).data
-
-      const { code } = compileTemplate({
-        id: JSON.stringify(id),
-        source: optimizedSvg,
-        transformAssetUrls: false
-      })
-
-      return `${code}\nexport default render`
+      const [path, query] = id.split('?', 2)
+      if (query === 'url') {
+        return path
+      } else if (query === 'raw') {
+        return await fs.readFile(path, 'utf-8')
+      }
+    },
+    async transform (src, id) {
```

Comment:
Could you explain why you moved this code from `load()` to `transform()`?

Reply:
In watch mode, the result of the `transform` hook is cached when rebuilding and the hook is only triggered again for a module id if either the code of the module has changed or a file has changed. 
[https://rollupjs.org/guide/en/#transform](url)